gnosis/safe-react
# fix: decouple pairing provider

## What it solves
Resolves #3669

## How this PR fixes it
The pairing provider has been decoupled from the pairing module. Interfacing with the pairing provider can now be done directly and doesn't use onboard as a proxy. We now: enable and disable connections, as well as retrieve the QR code from the provider.

## How to test it
- Open/close the wallet popper. The QR code should render (the same code each time).
- Pair via the QR code. Disconnect **VIA THE SAFE** and regenerate the QR code. It should work/allow you to connect again.
- Pair via the QR code. Disconnect **VIA THE APP** and regenerate the QR code. It should work/allow you to connect again.
- Rejecting the pairing request will regenerate a new QR code.
- Connect to MetaMask and disconnect. The QR code should generate correctly and allow for pairing.
- If the pairing module wasn't the last used provider, `Safe Mobile` should no longer populate the `provider.name` by default in Redux.
- Interrupted sessions should open with the Safe.
